File: server/models/problem.js
const mongoose = require("mongoose");

const sample = new mongoose.Schema({
  input: {
    type: String,
    required: true,
  },
  output: {
    type: String,
    required: true,
  },
});

const ProblemSchema = new mongoose.Schema({
  name: {
    type: String,
    required: true,
    unique: true,
  },
  author: {
    type: String,
    required: true,
  },
  statement: {
    type: String,
    required: true,
  },
  sampleTestcase: {
    type: [sample],
    required: true,
  },
  explanation: {
    type: String,
  },
  tags: {
    type: Array,
    required: true,
  },
  countAC: {
    type: Number,
    default: 0,
  },
  countTotal: {
    type: Number,
    default: 0,
  },
  systemTestcase: {
    type: [sample],
    required: true,
  },
  time: {
    type: Number,
    required: true,
  },
  memory: {
    type: Number,
    required: true,
  },
});

const Problem = mongoose.model("Problem", ProblemSchema);

module.exports = Problem;

File: src/main/java/com/onlinejudge/model/Problem.java
package com.onlinejudge.model;

import java.util.ArrayList;
import java.util.Map;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;


@Document(collection="problems")
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Data
public class Problem {
	
	@Id
	private String id;
	
	@Field("tags")
	private ArrayList<String> tags;
	
	@Field("countAC")
	private int countAC;
	
	@Field("countTotal")
	private int countTotal;
	
	@Field("name")
	private String name;
	
	@Field("author")
	private String author;
	
	@Field("statement")
	private String statement;
	
	@Field("explanation")
	private String explanation;
	
	@Field("sampleTestcase")
	private ArrayList<Map<String, ?>> sampleTestcase;
	
	@Field("systemTestcase")
	private ArrayList<Map<String, ?>> systemTestcase;
	
	@Field("time")
	private int time;
	
	@Field("memory")
	private int memory;
	
}
